samples/rtos/threadx-demo/main.cpp

```diff
@@ -103,194 +80,219 @@ int main() {
+                                                             DEMO_STACK_SIZE,
+                                                             1,
+                                                             1,
+                                                             MS_TO_TICKS(50),
+                                                             true);
+
+    // create the structs that holds the other thread arguments
+    other_thread_args_t thread_1_args = {&q1, &semaphore, &uarttx, &eventFlags, 1, &counters};
+
+    other_thread_args_t thread_2_args = {&q1, &semaphore, &uarttx, &eventFlags, 2, &counters};
+
+    other_thread_args_t thread_3_args = {&q1, &semaphore, &uarttx, &eventFlags, 3, &counters};
+
+    // create thread1
+    rtos::Thread<other_thread_args_t*> thread1(
+        (char*) "Thread 1", otherThreadEntry, &thread_1_args, DEMO_STACK_SIZE, 1, 1, MS_TO_TICKS(50), true);
+    // create thread2
+    rtos::Thread<other_thread_args_t*> thread2(
+        (char*) "Thread 2", otherThreadEntry, &thread_2_args, DEMO_STACK_SIZE, 1, 1, MS_TO_TICKS(50), true);
+    // create thread3
+    rtos::Thread<other_thread_args_t*> eventFlagThread(
+        (char*) "Thread 3", eventFlagThreadEntry, &thread_3_args, DEMO_STACK_SIZE, 1, 1, MS_TO_TICKS(50), true);
+
+    uart.printf("About to start the kernel.\n\r");
+
+    /*
+     * Most RTOS objects are initializable, which means we need to register them with the threadx kernel for them
+     * to function. This is done by putting all these initializable objects into an array (see below),
+     * and then passing that array into the startKernel method.
+     * If an object neeeds to be created and initialized after the kernel has been started, it's initialize method must
+     * be manually called. Also, make sure the length of the array is correct in startKernel.
+     */
+    rtos::Initializable* arr[] = {
+        &controllerThread, &uarttx, &q1, &semaphore, &eventFlags, &thread1, &thread2, &eventFlagThread};
+
+    // start the kernel (the kernel takes in the array of initializables and initializes them when the threadx kernel
+    // starts)
+    rtos::startKernel(arr, 8, txPool);
+
+    // the startKernel method doesn't actually return so this will never happen.
     return 0;
 }
 
-ULONG global_count  = 0; // Times a random number has been sent and received
-ULONG global_sum    = 0; // Sum of random numbers
-ULONG thread0_count = 0;
-ULONG thread0_sum   = 0;
-ULONG thread1_count = 0;
-ULONG thread1_sum   = 0;
-ULONG thread2_count = 0;
-ULONG thread2_sum   = 0;
-
-/// Function declarations
-void thread_0_entry(ULONG thread_input) {
-    // Setup UART
-    io::UART& uart = io::getUART<io::Pin::UART_TX, io::Pin::UART_RX>(9600);
+// Function definitions
 
+/**
+ * Controller Thread Entry Function, generates a random number then waits for the semaphore and writes stats to UART.
+ * @param args the argument struct this thread needs.
+ */
+void controllerThreadEntry(controller_thread_args_t* args) {
```

Comment:
Technically this doesn't generate this other threads, so I think controller thread makes more sense. At least to me generator makes it seem like it makes other threads. 
Also I'm not married to entry function as a terminology, we could use something like threadMain, since they're basically a main function the thread runs, and people are already familiar with what a main function is.

Reply:
I did call it generatorThread, because it generates random numbers.